datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/DomainUtils.java

```diff
@@ -33,4 +45,49 @@ public static boolean isAuthorizedToUpdateDomainsForEntity(@Nonnull QueryContext
         entityUrn.toString(),
         orPrivilegeGroups);
   }
+
+  public static void setDomainForResources(
+      @Nullable Urn domainUrn,
+      List<ResourceRefInput> resources,
+      Urn actor,
+      EntityService entityService
+  ) throws Exception {
+    final List<MetadataChangeProposal> changes = new ArrayList<>();
+    for (ResourceRefInput resource : resources) {
+      changes.add(buildSetDomainProposal(domainUrn, resource, actor, entityService));
+    }
+    ingestChangeProposals(changes, entityService, actor);
+  }
+
+  private static MetadataChangeProposal buildSetDomainProposal(
+      @Nullable Urn domainUrn,
+      ResourceRefInput resource,
+      Urn actor,
+      EntityService entityService
+  ) {
+    Domains domains = (Domains) getAspectFromEntity(
+        resource.getResourceUrn(),
+        Constants.DOMAINS_ASPECT_NAME,
+        entityService,
+        new Domains());
+    final UrnArray newDomains = new UrnArray();
+    if (domainUrn != null) {
+      newDomains.add(domainUrn);
+    }
+    domains.setDomains(newDomains);
```

Comment:
do we want to allow users to set multiple domains on an asset?

Reply:
Currently the product only allows one. The underlying data model was designed to support extension should this be a high-demand ask